gameplay/src/Matrix.cpp

```diff
@@ -906,12 +906,12 @@ void Matrix::transformPoint(const Vector3& point, Vector3* dst) const
 void Matrix::transformVector(Vector3* vector) const
 {
     GP_ASSERT(vector);
-    transformVector(vector->x, vector->y, vector->z, 0.0f, vector);
+    transformVector(vector->x, vector->y, vector->z, 1.0f, vector);
```

Comment:
I don't think this method will be very useful on the Matrix class. Unity's `Transform.TransformDirection` method is probably fairly fast because the scale, rotation and translate are likely stored separately in the Transform class. It would be easy to only apply the rotation to the vector. If you want to add a transformDirection method, it might be better to add it to the Transform class. 

We currently use `Matrix::transformVector(Vector3* vector)` in the Ray class but we don't care about the length of the direction vector in that case. I don't like how `Matrix::transformDirection` performs two square roots when people probably won't care about preserving the length of the direction vector in a lot of cases.